File: src/BreakfastProvider.Api/Models/Requests/CustomerPreferenceRequest.cs
namespace BreakfastProvider.Api.Models.Requests;

public record CustomerPreferenceRequest
{
    public string? CustomerId { get; init; }
    public string? CustomerName { get; init; }
    public string? PreferredMilkType { get; init; }
    public bool LikesExtraToppings { get; init; }
    public string? FavouriteItem { get; init; }
}

File: java/breakfast-provider-app/src/main/java/io/lemonlion/breakfast/model/request/CustomerPreferenceRequest.java
package io.lemonlion.breakfast.model.request;

/** Twin of C# {@code CustomerPreferenceRequest}. */
public record CustomerPreferenceRequest(
        String customerId,
        String customerName,
        String preferredMilkType,
        boolean likesExtraToppings,
        String favouriteItem) {

    /** Returns a copy with the customer id set from the path (C# does {@code request with { CustomerId = ... }}). */
    public CustomerPreferenceRequest withCustomerId(String id) {
        return new CustomerPreferenceRequest(id, customerName, preferredMilkType, likesExtraToppings, favouriteItem);
    }
}
